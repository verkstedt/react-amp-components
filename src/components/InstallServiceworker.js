import React, { Component, Fragment } from 'react'
import { string } from 'prop-types'
import { Helmet } from 'react-helmet'

import pick from '../utils/pick'
import mapKeys from '../utils/mapKeys'
import commonPropTypes from './commonPropTypes'

class InstallServiceWorker extends Component {
  static propTypes = {
    src: string.isRequired,
    dataIframeSrc: string,
    dataNoServiceWorkerFallbackUrlMatch: string,
    dataNoServiceWorkerFallbackShellUrl: string
  }

  static defaultProps = {
    dataIframeSrc: undefined,
    dataNoServiceWorkerFallbackUrlMatch: undefined,
    dataNoServiceWorkerFallbackShellUrl: undefined
  }

  get properties() {
    return mapKeys(pick(this.props,
      'src',
      'dataIframeSrc',
      'dataNoServiceWorkerFallbackUrlMatch',
      'dataNoServiceWorkerFallbackShellUrl'
    ))
  }

  render() {
    return (
      <Fragment>
        <Helmet key="helmet">
          <script async custom-element="amp-install-serviceworker" src="https://cdn.ampproject.org/v0/amp-install-serviceworker-0.1.js" />
        </Helmet>
        <amp-install-serviceworker
          key="install-serviceworker"
          {...this.properties}
          layout="nodisplay"
        />
      </Fragment>
    )
  }
}

export default InstallServiceWorker
