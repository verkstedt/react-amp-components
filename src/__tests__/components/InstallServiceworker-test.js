import React from 'react'
import InstallServiceworker from '../../components/InstallServiceworker'
import Helmet from '../../utils/Helmet'
import { renderComponent } from '../test-utils'

describe('InstallServiceworker', () => {
  it('matches the snapshot', () => {
    const installServiceworker = renderComponent(
      <InstallServiceworker
        src="src"
        dataIframeSrc="dataIframeSrc"
        dataNoServiceWorkerFallbackUrlMatch="test"
      />
    )
    expect(installServiceworker.toJSON()).toMatchSnapshot()
  })

  it('matches the snapshot with no optional props set', () => {
    const installServiceworker = renderComponent(
      <InstallServiceworker src="src" />
    )
    expect(installServiceworker.toJSON()).toMatchSnapshot()
  })

  it('should correctly set layout prop', () => {
    const installServiceworker = renderComponent(
      <InstallServiceworker src="src" layout="not-correct" />
    )
    expect(installServiceworker.toJSON()).toMatchSnapshot()
  })

  it('injects the right script tag', () => {
    renderComponent(
      <InstallServiceworker />
    )
    Helmet.canUseDOM = false
    const staticHead = Helmet.renderStatic()
    expect(staticHead.scriptTags).toMatchSnapshot()
  })
})
