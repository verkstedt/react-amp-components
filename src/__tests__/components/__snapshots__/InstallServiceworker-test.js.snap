// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallServiceworker injects the right script tag 1`] = `
Array [
  Object {
    "async": true,
    "custom-element": "amp-install-serviceworker",
    "src": "https://cdn.ampproject.org/v0/amp-install-serviceworker-0.1.js",
  },
]
`;

exports[`InstallServiceworker matches the snapshot 1`] = `
<amp-install-serviceworker
  data-iframe-src="dataIframeSrc"
  data-no-service-worker-fallback-url-match="test"
  layout="nodisplay"
  src="src"
/>
`;

exports[`InstallServiceworker matches the snapshot with no optional props set 1`] = `
<amp-install-serviceworker
  layout="nodisplay"
  src="src"
/>
`;

exports[`InstallServiceworker should correctly set layout prop 1`] = `
<amp-install-serviceworker
  layout="nodisplay"
  src="src"
/>
`;
